api/client/ffs.go

```diff
@@ -92,15 +118,22 @@ func (f *ffs) Info(ctx context.Context) (*pb.InfoReply, error) {
 	return f.client.Info(ctx, &pb.InfoRequest{})
 }
 
-func (f *ffs) Watch(ctx context.Context, jids ...ff.JobID) (<-chan JobEvent, error) {
+func (f *ffs) Watch(ctx context.Context, jids ...ff.JobID) (<-chan JobEvent, func(), error) {
```

Comment:
@jsign you mentioned how it's a bit of a strange API to require the user to cancel a context in order to clean up. This is a slight improvement where we return a cancel function they can call.

Reply:
My original mention was related to adding comments to point out that the API is expecting that (call `cancel()`). That applies also to this change since someone might not know about what is that second `func()` returned and what is it for.

To be clear: I wasn't really proposing a change, I think both approaches are OK. It's just to clarify what's expected by the client in docs.

Reply:
Gotcha.